include/mapnik/metrics.hpp

```diff
@@ -48,6 +46,35 @@ enum measurement_t : int_fast8_t
     TOTAL_ENUM_SIZE
 };
 
+#if __has_cpp_attribute(maybe_unused)
+#define METRIC_UNUSED [[maybe_unused]]
+#elif __has_cpp_attribute(gnu::unused)
+#define METRIC_UNUSED [[gnu::unused]]
+#else
+#define METRIC_UNUSED
+#endif
```

Comment:
I don't quite understand how this macro works. Could you please explain it to me? (or point to some documentation).

At the same time, I'm a bit wary of breaking mac builds (which I cannot test in travis right now).

Reply:
The whole idea with the PR is to not need guards anymore by setting two options with the metrics: the full functionality if the option is enabled or a dummy one simple enough to be (hopefully) removed automatically by the compiler if not.

This macro is  used to remove warnings of unused variable declarations if the code is compiled without metrics support. It uses cpp++17 attribute if avaible, if not tries to use a gcc attribute (also available in clang) and if not it just doesn't do anything and the warning will be shown.